homeassistant/components/overkiz/select.py

```diff
@@ -0,0 +1,134 @@
+
+from . import HomeAssistantOverkizData
+from .const import DOMAIN, IGNORED_OVERKIZ_DEVICES
+from .entity import OverkizDescriptiveEntity, OverkizDeviceClass
+
+
+@dataclass
+class OverkizSelectDescriptionMixin:
+    """Define an entity description mixin for select entities."""
+
+    options: list[str | OverkizCommandParam]
+    select_option: Callable[[str, Callable[..., Awaitable[None]]], Awaitable[None]]
+
+
+@dataclass
+class OverkizSelectDescription(SelectEntityDescription, OverkizSelectDescriptionMixin):
+    """Class to describe an Overkiz select entity."""
+
+
+def _select_option_open_closed_pedestrian(
+    option: str, execute_command: Callable[..., Awaitable[None]]
+) -> Awaitable[None]:
+    """Change the selected option for Open/Closed/Pedestrian."""
+    return execute_command(
+        {
+            OverkizCommandParam.CLOSED: OverkizCommand.CLOSE,
+            OverkizCommandParam.OPEN: OverkizCommand.OPEN,
+            OverkizCommandParam.PEDESTRIAN: OverkizCommand.SET_PEDESTRIAN_POSITION,
+        }[OverkizCommandParam(option)],
+        None,
+    )
+
+
+def _select_option_memorized_simple_volume(
+    option: str, execute_command: Callable[..., Awaitable[None]]
+) -> Awaitable[None]:
+    """Change the selected option for Memorized Simple Volume."""
+    return execute_command(OverkizCommand.SET_MEMORIZED_SIMPLE_VOLUME, option)
+
+
+SELECT_DESCRIPTIONS: list[OverkizSelectDescription] = [
+    OverkizSelectDescription(
+        key=OverkizState.CORE_OPEN_CLOSED_PEDESTRIAN,
+        name="Position",
+        icon="mdi:content-save-cog",
+        options=[
+            OverkizCommandParam.OPEN,
+            OverkizCommandParam.PEDESTRIAN,
+            OverkizCommandParam.CLOSED,
+        ],
+        select_option=_select_option_open_closed_pedestrian,
+        device_class=OverkizDeviceClass.OPEN_CLOSED_PEDESTRIAN,
+    ),
+    OverkizSelectDescription(
+        key=OverkizState.IO_MEMORIZED_SIMPLE_VOLUME,
+        name="Memorized Simple Volume",
+        icon="mdi:volume-high",
+        options=[OverkizCommandParam.STANDARD, OverkizCommandParam.HIGHEST],
+        select_option=_select_option_memorized_simple_volume,
+        entity_category=EntityCategory.CONFIG,
+        device_class=OverkizDeviceClass.MEMORIZED_SIMPLE_VOLUME,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Overkiz select from a config entry."""
+    data: HomeAssistantOverkizData = hass.data[DOMAIN][entry.entry_id]
+    entities: list[OverkizSelect] = []
+
+    key_supported_states = {
+        description.key: description for description in SELECT_DESCRIPTIONS
+    }
+
+    for device in data.coordinator.data.values():
+        if (
+            device.widget in IGNORED_OVERKIZ_DEVICES
+            or device.ui_class in IGNORED_OVERKIZ_DEVICES
+        ):
+            continue
+
```

Comment:
This pattern exists in a few places. Can this be more DRY?

Reply:
Would this be blocking this current PR? I noted it on the list for refactors / feedback that we need to address. 

I would love to have `select` and (later on) `switch` in, so that all platforms that share some of the same logic are in. Than we can refactor the `async_setup_entry` to be more DRY.

Reply:
It can be done in a follow-up